src/components/LinkChooser/SearchInput.tsx

```diff
@@ -0,0 +1,178 @@
+        const { isFocusVisible: copyButtonIsFocusVisible, focusProps: copyButtonFocusProps } = useFocusRing();
+        const { isFocusVisible: clearButtonIsFocusVisible, focusProps: clearButtonFocusProps } = useFocusRing();
+
+        const [, send] = useActor(machineService);
+
+        const onKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+            if (event.key === "Enter") {
+                onEnterPressed && onEnterPressed(event);
+            }
+        };
+
+        return (
+            <div
+                {...focusProps}
+                data-test-id="link-chooser-search-wrapper"
+                className={merge([
+                    "tw-flex tw-items-center tw-h-9 tw-gap-2 tw-px-3 tw-border tw-rounded tw-text-s tw-font-sans tw-relative tw-bg-white dark:tw-bg-transparent",
+                    dotted ? "tw-border-dashed" : "tw-border-solid",
+                    disabled
+                        ? "tw-border-black-5 tw-bg-black-5 dark:tw-bg-black-90 dark:tw-border-black-90"
+                        : merge([
+                              "focus-within:tw-border-black-90",
+                              validationStyle[validation],
+                              isFocusVisible && FOCUS_STYLE,
+                          ]),
+                ])}
+            >
+                {decorator && (
+                    <div
+                        className={merge([
+                            "tw-flex tw-items-center tw-justify-center tw-pl-1",
+                            disabled ? "tw-text-black-60" : "tw-text-black-80",
+                        ])}
+                        data-test-id="link-chooser-decorator-icon"
+                    >
+                        {decorator}
+                    </div>
+                )}
+                <input
+                    id={useMemoizedId(propId)}
+                    ref={inputElement}
+                    className={merge([
+                        "tw-flex-grow tw-border-none tw-outline-none tw-bg-transparent tw-hide-input-arrows",
+                        disabled
+                            ? "tw-text-black-40 tw-placeholder-black-30 dark:tw-text-black-30 dark:tw-placeholder-black-40"
+                            : "tw-text-black tw-placeholder-black-60 dark:tw-text-white",
+                    ])}
+                    onChange={onChange}
+                    onFocus={onFocus}
+                    onBlur={onBlur}
+                    onKeyDown={onKeyDown}
+                    onClick={onClick}
+                    placeholder={placeholder}
+                    value={value}
+                    type={type}
+                    required={required}
+                    disabled={disabled}
+                    size={size}
+                    data-test-id="link-chooser-search-input"
+                />
+                {selectedResult && previewable && (
+                    <button
```

Comment:
the buttons seem quite similar => maybe use it's own component for it, that way you don't have to repeat useFocus states etc.

Reply:
done 👍